main.go

```diff
@@ -210,24 +207,53 @@ func newPrometheusServer(metricsAddr string, cache cache.Cache) (*prometheusServ
 	}, nil
 }
 
+func extractHostnameFromCert(certPath string) (string, error) {
+	certBytes, err := os.ReadFile(certPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read certificate file: %w", err)
+	}
+
+	block, _ := pem.Decode(certBytes)
+	if block == nil {
+		return "", fmt.Errorf("failed to parse certificate PEM")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse certificate: %w", err)
+	}
+
+	if cert.Subject.CommonName != "" {
+		return cert.Subject.CommonName, nil
+	}
+
+	if len(cert.DNSNames) > 0 {
+		return cert.DNSNames[0], nil
+	}
+
+	return "", fmt.Errorf("no hostname found in certificate")
+}
+
 func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
+	var olmDeployment bool
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8443", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&olmDeployment, "olm-deployment", false, "Signalize deployment via OLM")
 	opts := zap.Options{}
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	err := createCertificateSymlinks()
```

Comment:
I see you've removed the symlinks. Are they not needed anymore?
If yes, can you do it in a separate commit, because it is not related to the metrics change.